### ISSUE TITLE
set distanceFilter and stationaryRadius causes location event to fire in loop

### ISSUE THREAD

**Issue by angel1st** [mauron85/cordova-plugin-background-geolocation#485]
## Your Environment

* Plugin version:
3.0.0-alpha.41
* Platform: Android
* OS version:
8.0.0 / 7.0.0
* Device manufacturer and model:
S8 / Redmi Note4
* Cordova version (`cordova -v`):
8.0.0
* Cordova platform version (`cordova platform ls`):
android 7.1.0
ios 4.5.5
* Plugin configuration options: 
```
{
                        locationProvider: BackgroundGeolocation.ACTIVITY_PROVIDER,
                        desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
                        stationaryRadius: distance,
                        distanceFilter: distance,
                        notificationTitle: 'TrackMe is tracking',
                        notificationText: 'enabled',
                        debug: false,
                        interval: 10000,
                        fastestInterval: 5000,
                        activitiesInterval: 10000,
                        stopOnTerminate: true,
                        startOnBoot: false
                    }
```
* Link to your project:

## Context

I would like to set `distanceFilter` dynamically based on the `speed`. For the purpose, I would like to use [the speed calculation formula](https://github.com/socrates3/cordova-plugin-background-geolocation#distancefilter).
How I am trying to achieve this:
1. I have defined following method:

```
recalculateDistanceFilter = speed => {
                        speed = speed || 0;
                        let newDistanceFilter = Math.pow((Math.round(speed, 5)), 2) + distance;
                        BackgroundGeolocation.configure({
                            stationaryRadius: newDistanceFilter,
                            distanceFilter: newDistanceFilter,
                        });
                };
```
1. I am calling the method from within `location` event listener:

```
BackgroundGeolocation.on('location', (location) => {
                        // handle your locations here
                        // to perform long running operation on iOS
                        // you need to create background task

                        self.recalculateDistanceFilter(location.speed);

                        BackgroundGeolocation.startTask((taskKey) => {
                            // execute long running task
                            // eg. ajax post location
                            // IMPORTANT: task has to be ended by endTask
                            BackgroundGeolocation.endTask(taskKey);
                        });
                    });
```

## Expected Behavior

The above method call should simply set new values for both `distanceFilter` and `stationaryRadius`. The next time the new threshold is reached, it will fire `location` event.

## Actual Behavior

Every time:

```
BackgroundGeolocation.configure({
                            stationaryRadius: newDistanceFilter,
                            distanceFilter: newDistanceFilter,
                        });
```
is called, it fires `location` event and the app gets into a viscous circle, since within `location` event `BackgroundGeolocation.configure` is called, it somehow fires `location` event and so forth.
The loop stops when `activity` event fires with `STILL` type, however, when device moves again and `location` event fires, the viscous circle starts again.
So basically, instead of reduce number of location request, they increased tremendously.  

 
## Steps to Reproduce

1. Call 

```
BackgroundGeolocation.configure({
                            stationaryRadius: newDistanceFilter,
                            distanceFilter: newDistanceFilter,
                        });
```
within `location` event listener
2.
3.
4.

## Context

I would like to reduce the number of location requests in case speed increases and vise versa.

**Comment by jfbourne** [comment]
I would like to achieve the same, if user is walking I would like to set different stationaryRadius than when user is driving for instance? @mauron85 is this possible, if so would you please share your knowledge? Most important thank you for your hard work.
